Add Krux project page (#765)
OpenSats/website: data/projects/krux.mdx

---
title: 'Krux'
dateAdded: '2026-05-17'
summary: 'Open-source firmware that turns inexpensive devices into air-gapped Bitcoin signing tools.'
nym: 'Odudex and Tadeubas'
website: 'https://selfcustody.github.io/krux/'
coverImage: '/static/images/projects/krux-logo.png'
git: 'https://github.com/selfcustody/krux'
tags: ['Bitcoin', 'Wallet', 'Signing']
fund: general
announcementLink: '/blog/bitcoin-grants-december-2023#krux'
---

> Krux is open-source firmware that turns inexpensive Kendryte K210 boards into air-gapped Bitcoin signing devices. It supports single-signature and multisignature wallets, moves [PSBTs][psbt] over QR codes or SD cards, and includes tools for message signing, backups, and offline wallet review.

Reviewer: ```suggestion
Krux is open-source firmware that turns inexpensive [Kendryte K210][k210] boards into air-gapped Bitcoin signing devices. It supports single-signature and multisignature wallets, moves [PSBTs][psbt] over QR codes or SD cards, and includes tools for message signing, backups, and offline wallet review.
```

Recent public work has pushed Krux well beyond a basic signer. The project now supports more devices, clearer descriptor handling, stronger offline backup tooling, and regular security hardening across its QR, PSBT, and SD-card flows.

## Why fund it?

Air-gapped self-custody should be open, inspectable, and affordable. Krux lowers the cost of secure signing and gives users a firmware stack they can audit, improve, and adapt to locally available hardware.

That matters even more in places where import fees can make commercial hardware wallets expensive. OpenSats first supported Krux in the [third wave of Bitcoin grants][grant] and has continued backing the project as it expanded device support, coordinator compatibility, and signer safety.

## What's next?

Recent public updates show Krux pushing on three fronts: broader hardware support, smoother signer UX, and tighter security. Releases through [v26.04.0][release] added more supported devices, improved descriptor and QR handling, and shipped a broad set of safety checks around PSBT parsing, KEF decryption, and SD-card inputs.

> The public roadmap is still moving toward faster QR tooling, continued Kapps work, and more polished device flows. The [OpenSats open-hardware report][report] is a good snapshot of where the project has been heading across firmware releases, new device ports, and ongoing signer UX work.

Reviewer: ```suggestion
The public roadmap is still moving toward faster QR tooling, continued Kapps work, and more polished device flows. The [Open Hardware for Open Money report][report] is a good snapshot of where the project has been heading across firmware releases, new device ports, and ongoing signer UX work.
```

Consistent with BitAxe project page

If you want to dig deeper, start with the [project site][site], the [main repository][repo], the [grant announcement][grant], or the [release history][releases].

[grant]: /blog/bitcoin-grants-december-2023#krux
[installer]: https://github.com/selfcustody/krux-installer/releases

> [psbt]: https://bitcoinops.org/en/topics/psbt/

Reviewer: ```suggestion
[psbt]: /topics/psbt/
```

[release]: https://github.com/selfcustody/krux/releases/tag/v26.04.0
[releases]: https://github.com/selfcustody/krux/releases
[report]: /blog/open-hardware-for-open-money#krux
[repo]: https://github.com/selfcustody/krux

> [site]: https://selfcustody.github.io/krux/

Reviewer: ```suggestion
[site]: https://selfcustody.github.io/krux/
[k210]: https://www.kendryte.com/en/proDetail/210
```
